src/cargo/core/workspace.rs

```diff
@@ -991,11 +980,38 @@ impl<'cfg> Workspace<'cfg> {
                 if !manifest.patch().is_empty() {
                     emit_warning("patch")?;
                 }
-                if let Some(behavior) = manifest.resolve_behavior() {
-                    if behavior != self.resolve_behavior {
-                        // Only warn if they don't match.
-                        emit_warning("resolver")?;
+                if manifest.defaulted_resolve_behavior() != self.resolve_behavior {
+                    let package_reason = manifest
```

Comment:
> any inconsistency in workspace and package resolver 

imo this doesn't lead to a good user experience for someone explicitly setting the resolver in the workspace while leaving it defaulted in workspace members.

While its unlikely for someone to be in this place with v2 and edition = "2015", I doubt v2 is the last version of the resolver we'll have.

Reply:
The only time I would expect there to be much churn from this is during a piecemeal migration to a new edition which implies a new resolver. Then you would need to explicitly either set the unmigrated crates forward to the new resolver, or set the migrated crates back to the old resolver. For an all-at-once migration you would just bump the `edition` in every package and the `resolver` in the root and everything would be good.

Reply:
(And of course during a piecemeal migration you could just ignore the warnings until the entire workspace has been migrated to avoid that churn).